packages/react/src/components/hooks/usePermissions.ts

```diff
@@ -0,0 +1,74 @@
+import { useEffect, useState } from "react";
+import { Video } from "@signalwire/js";
+import jwt_decode from "jwt-decode";
+
+function makeBarePermissionObject(permString: string[]) {
+  let permObject = {};
+  let permElements = permString.map((perm) => perm.split(".").splice(1));
+  permElements.forEach((perm) => {
+    addPerm(perm, permObject);
+    function addPerm(perm: string[], permObj: any) {
+      if (perm.length === 1) permObj[perm[0]] = true;
+      else {
+        if (permObj[perm[0]] === undefined) permObj[perm[0]] = {};
+        addPerm(perm.splice(1), permObj[perm[0]]);
+      }
+    }
+  });
+  let handler: ProxyHandler<any> = {
+    get(target, prop) {
+      if (typeof target[prop] === "object")
+        return new Proxy(target[prop], handler);
+      else if (target[prop] === undefined) return false;
+      return target[prop];
+    },
+  };
+  let proxyPerm = new Proxy(permObject, handler);
+  return proxyPerm;
+}
+
+function decoratePermissionObject(P_bare: any) {
+  if (!(typeof P_bare === "object")) return {};
+  let P = JSON.parse(JSON.stringify(P_bare)); //Deep copy for small object
+
+  P.self.audio = P.self.audio_mute && P.self.audio_unmute;
+  P.self.video = P.self.video_mute && P.self.video_unmute;
+  P.self.speaker = P.self.deaf && P.self.undeaf;
+  P.self.remove = true;
```

Comment:
I like this solution of decorating the permission object to add the shortcuts.

Do you think we could find a nice prefix for those, to avoid any future clashing with upstream names? Something like `P.self.all_audio`, `P.self.all_video`, even though I'm not much convinced by `all_` specifically. Another possibility could be `P.self.audio.all`.

Reply:
I wouldn't go with `P.self.audio.all` unless we also did something like `P.self.audio.mute/unmute` but maybe `P.self.audio_full`? matches with `audio_muted/unmuted`

Reply:
`audio_full` sounds good to me